src/main/kotlin/com/jetbrains/snakecharm/codeInsight/SnakemakeParamsInShellReferenceContributor.kt

```diff
@@ -26,12 +26,17 @@ class SnakemakeParamsInShellReferenceContributor : PsiReferenceContributor() {
                             context: ProcessingContext
                     ): Array<PsiReference> {
                         val paramReferences = mutableListOf<PsiReference>()
-
                         val paramsMatcher = paramsPattern.matcher(element.text)
 
-                        while (paramsMatcher.find()) {
-                            paramReferences.add(SMKParamsReference(element,
-                                    TextRange(paramsMatcher.start(1), paramsMatcher.end(1))))
+                        val isShellCommand = PsiTreeUtil
+                                .getParentOfType(element, SMKRuleParameterListStatement::class.java)
+                                ?.section
```

Comment:
Could `.getParentOfType(element, SMKRuleParameterListStatement::class.java)` be null here?

Reply:
the question is discussed in another comment

Reply:
No it could not, but I have tried removing this safe call and code didn't compile, as I have mentioned above.